angel-ps/core/src/main/java/com/tencent/angel/ml/math2/matrix/CooDoubleMatrix.java

```diff
@@ -0,0 +1,121 @@
+package com.tencent.angel.ml.math2.matrix;
+
+import com.tencent.angel.ml.math2.storage.IntDoubleSparseVectorStorage;
+import com.tencent.angel.ml.math2.vector.IntDoubleVector;
+import com.tencent.angel.ml.math2.vector.Vector;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CooDoubleMatrix extends CooMatrix{
+
+  protected double [] values;
+
+  public CooDoubleMatrix() {
+    super();
+  }
+
+  public CooDoubleMatrix(int matrixId, int clock, int[] rowIndices, int[] colIndices, double[] values, int[] shape) {
+    this.matrixId = matrixId;
+    this.clock = clock;
+    this.rowIndices = rowIndices;
+    this.colIndices = colIndices;
+    this.values = values;
+    this.shape = shape;
+  }
+
+  public CooDoubleMatrix(int[] rowIndices, int[] colIndices, double[] values, int[] shape) {
+    this(0, 0, rowIndices, colIndices, values, shape);
+  }
+
+  @Override
+  public Vector getRow(int idx) {
+    List cols = new ArrayList<Integer>();
```

Comment:
replacing ArrayList with IntArrayList/FloatArrayList of fastutils

Reply:
ok